org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/guice/AbstractGuiceAwareExecutableExtensionFactory.java

```diff
@@ -49,15 +51,44 @@ public Object create() throws CoreException {
 		try {
 			final Class<?> clazz = getBundle().loadClass(clazzName);
 			final Injector injector = getInjector();
+			if (injector == null) {
+				throw handleCreationError(null);
+			}
 			final Object result = injector.getInstance(clazz);
 			if (result instanceof IExecutableExtension)
 				((IExecutableExtension) result).setInitializationData(config, null, null);
 			return result;
+		} catch (VirtualMachineError e) {
+			throw e;
+		} catch (Throwable e) {
+			throw handleCreationError(e);
 		}
-		catch (Exception e) {
-			log.error(e.getMessage(), e);
-			throw new CoreException(new Status(IStatus.ERROR, getBundle().getSymbolicName(), e.getMessage() + " ExtensionFactory: "+ getClass().getName(), e));
+	}
+	
+	/**
+	 * In the case that the EEF fails to create an injector or an instance of the desired class
+	 * this method is called to produce a more user friendly error message.
+	 * @since 2.13
+	 */
+	protected CoreException handleCreationError(Throwable cause) {
+		Bundle bundle = null;
+		String bundleName = null;
+		try {
+			bundle = getBundle();
+		} catch (NullPointerException e) {
+			// subclasses generated with Xtext <2.13 will raise a NPE when the UI bundle was not activated
+			// try to guess the bundle from the EEF's package
+			bundle = Platform.getBundle(getClass().getPackage().getName());
 		}
+		bundleName = bundle != null ? bundle.getSymbolicName() : "UNKNOWN";
+
+		if (bundle != null && bundle.getState() != Bundle.ACTIVE) {
+			return new CoreException(new Status(IStatus.ERROR, bundleName, "Bundle " + bundleName
+					+ " not activated. Please check that all dependencies could be resolved and 'Bundle-ActivationPolicy: lazy' is configured in the bundle's MANIFEST.MF. Check also the error log."));
+		}
+
+		return new CoreException(
```

Comment:
yes, this is intended. The CoreException will be thrown and handled by the framework. The user can't work with the DSL UI, but gets the appropriate error message. And if enabled the exception gets reported to AERI.